### PR TITLE
[bench] Adding a cpp benchmark to compare performance of nnc with static and symbolic shapes

### PR DESCRIPTION
**Pull request by navahgar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72198
* __->__ #72197
* #72196

Benchmark results:
```
--------------------------------------------------------------------------
Benchmark                                Time             CPU   Iterations
--------------------------------------------------------------------------
KernelBench/Eager/32                  2007 ns         2007 ns       350630
KernelBench/Eager/64                  2050 ns         2050 ns       335326
KernelBench/Eager/512                 2196 ns         2196 ns       320889
KernelBench/Eager/2048                2853 ns         2853 ns       233259
KernelBench/StaticShapes/32            544 ns          544 ns      1303151
KernelBench/StaticShapes/64            559 ns          559 ns      1271121
KernelBench/StaticShapes/512           586 ns          586 ns      1149529
KernelBench/StaticShapes/2048          820 ns          820 ns       855455
KernelBench/SymbolicShapes/32          704 ns          704 ns      1008014
KernelBench/SymbolicShapes/64          730 ns          730 ns       981898
KernelBench/SymbolicShapes/512         727 ns          727 ns       893909
KernelBench/SymbolicShapes/2048       1065 ns         1064 ns       648401
```

Differential Revision: [D33951742](https://our.internmc.facebook.com/intern/diff/D33951742)

### CODE REVIEW

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951742).

File: benchmarks/cpp/tensorexpr/bench_kernels.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    TensorExprKernel k(
+        graph, {}, symbolic_shape_inputs, false, symbolic_strides);
+
+    auto stmt = k.getCodeGenStmt();
+    std::cout << "kernel stmt: " << *stmt << std::endl;
+
+    for (auto _ : state) {
+      auto a = at::rand({dim}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+      auto b = at::rand({dim}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+      std::vector<at::Tensor> inputs = {a, b};
```
Comment (huiguoo) [inline benchmarks/cpp/tensorexpr/bench_kernels.cpp:74]: Should we exclude the input preparation process from measurement as well? So static and symbolic are compared fairly.

File: benchmarks/cpp/tensorexpr/bench_kernels.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+BENCHMARK_DEFINE_F(KernelBench, StaticShapes)(benchmark::State& state) {
+  GraphWithStaticShapes(state);
+}
+BENCHMARK_DEFINE_F(KernelBench, SymbolicShapes)(benchmark::State& state) {
+  GraphWithSymbolicShapes(state);
+}
+
+BENCHMARK_REGISTER_F(KernelBench, StaticShapes)->Args({1500});
```
Comment (huiguoo) [inline benchmarks/cpp/tensorexpr/bench_kernels.cpp:90]: Is `1500` extracted from a real use case? Are we interested in benchmarking other dims like from 32 to 2048?

**Review by huiguoo — APPROVED** [review]
LGTM! 
I have two other comments below.

**Comment by eellison** [comment]
where do the results of the benchmark show up ?

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951742).

File: benchmarks/cpp/tensorexpr/bench_kernels.cpp (hunk 1 above)
Comment (navahgar) [reply benchmarks/cpp/tensorexpr/bench_kernels.cpp]: Good point. Moved.

**Comment by navahgar** [comment]
> where do the results of the benchmark show up ?

Added eager as well. Updated the PR description with the results.

**Pull request closed without merging**